std/json.ln

```diff
@@ -0,0 +1,633 @@
+/* @std/json - A library for parsing, representing, and serializing to and from JSON
+ * This representation can handle every JSON type, and if you squint at it, it cannot represent an
+ * invalid value. This is not entirely true without a small liberty being taken. If a JSONNode is an
+ * object, its children *should* be `KeyVal<string, JSONBase>` type, but if they're just `JSONBase`
+ * you could also just use the `toString` representation of the value as its key and the value as
+ * the value. Similarly, you should not ever have children attached to a `JSONBase`, but you can
+ * just ignore those children.
+ */
+export type IsObject {
+  isObject: bool,
+}
+// TODO: Fix issues with using `void` as a value in the compiler
```

Comment:
we should track this with a task if one doesn't exist

Reply:
I plan to make two tasks after this, at minimum. Not just `void` as value not working, but also the variable aliasing bug I ran across.